src/brsTypes/components/RoSGNode.ts

```diff
@@ -48,22 +50,28 @@ class Field {
     }
 
     setValue(value: BrsType) {
-        if (this.alwaysNotify || this.value !== value) {
+        let oldValue = this.value;
+        this.value = value;
+        if (this.alwaysNotify || oldValue !== value) {
             this.observers.map(this.executeCallbacks);
         }
-        this.value = value;
     }
 
     addObserver(interpreter: Interpreter, callable: Callable) {
         let brsCallback: BrsCallback = {
             interpreter: interpreter,
+            environment: interpreter.environment,
             callable: callable,
         };
         this.observers.push(brsCallback);
     }
 
     private executeCallbacks(callback: BrsCallback) {
-        callback.callable.call(callback.interpreter);
+        let { interpreter, callable, environment } = callback;
+        interpreter.inSubEnv(subInterpreter => {
+            callable.call(subInterpreter);
+            return BrsInvalid.Instance;
```

Comment:
Does TypeScript complain about not returning a `BrsType` here?  That's a bummer since we're not actually using the return value, but I suppose we also don't want to return the result of that `call` all the time. 🤔 

If this feels like a pain in the butt to use, I'm open to suggestions!

Reply:
Yep, it does complain about that. I think it works for now 🤷‍♂️